nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1204,8 +1213,9 @@ mod test {
     }
 
     #[test]
-    fn test_disk_expungement_removes_zones() {
-        static TEST_NAME: &str = "planner_disk_expungement_removes_zones";
+    fn test_disk_expungement_removes_zones_durable_zpool() {
```

Comment:
DId this name change because there are only durable zpools (crucible) for each disk with only a single zpool?

Reply:
The `zone_needs_expungement` function (which I modified above) has two different checks -- one for expunging zones using "durable datasets from zpools", and one for zones using "transient filesystem zpools".

Because zones using durable zpools *also* use transient filesystems -- and co-locate them -- these cases are slightly muddied. As you suggested, Crucible has a "durable" zpool, and it happens to pick the same transient zpool for a root filesystem.

I renamed the test to make it more clear what I'm testing:
- In this test case, I'm checking the "durable dataset removal" path.
- In the other test case, I pick a zone without a durable dataset to test the other pathway

Reply:
Great. Thanks for the details.